app/src/main/java/com/zeapo/pwdstore/crypto/PasswordCreationActivity.kt

```diff
@@ -223,9 +223,9 @@ class PasswordCreationActivity : BasePgpActivity(), OpenPgpServiceConnection.OnB
         val data = Intent()
         data.action = OpenPgpApi.ACTION_ENCRYPT
 
-        // EXTRA_KEY_IDS requires long[]
-        val longKeys = keyIDs.map { it.toLong() }
-        data.putExtra(OpenPgpApi.EXTRA_KEY_IDS, longKeys.toLongArray())
+        // pass enters the key ID into `.gpg-id`.
+        val keyIdFile = File(category.text.toString()).walkBottomUp().first { it.name == ".gpg-id" }
```

Comment:
I generally like this approach, but couldn't this crash if there is no `.gpg-id` in the current category, but only in the top-level directory?

Reply:
Nope, this traverses from the bottom to top until it finds a `.gpg-id` file so it'll always find the root key. I've verified that behavior with my personal store. Will check if we're able to add tests for this.

Reply:
I thought that this would only visit subfolders and files in `category.text.toString()`, which may not cover the top-level repo directory. But I will gladly let the tests prove me wrong.

BTW: Should we handle the case where the user deletes all the `.gpg-id` files externally? This would crash on `first`.

Reply:
Yeah I think we can account for that, I'll push a fix.